cobigen/cobigen-tsplugin/src/main/java/com/devonfw/cobigen/tsplugin/inputreader/TypeScriptInputReader.java

```diff
@@ -0,0 +1,289 @@
+     */
+    private ExternalProcessHandler request = ExternalProcessHandler
+        .getExternalProcessHandler(ExternalProcessConstants.HOST_NAME, ExternalProcessConstants.PORT);
+
+    /**
+     * Exception handler related to connectivity to the server
+     */
+    private ConnectionExceptionHandler connectionExc = new ConnectionExceptionHandler();
+
+    /**
+     * Creates a new {@link TypeScriptInputReader}
+     */
+    public TypeScriptInputReader() {
+
+        try {
+            // We first check if the server is already running
+            request.startConnection();
+            if (request.isNotConnected()) {
+                startServerConnection();
+            }
+        } catch (IOException e) {
+            // If it is not currently running, we need to execute it
+            LOG.info("Server is not currently running. Let's initialize it");
+            startServerConnection();
+        }
+
+    }
+
+    /**
+     * Deploys the server and tries to initialize a new connection between CobiGen and the server
+     */
+    private void startServerConnection() {
+        request.executingExe(Constants.EXE_NAME, this.getClass());
+        request.initializeConnection();
+    }
+
+    @Override
+    public boolean isValidInput(Object input) {
+        String fileContents = null;
+        String inputCharset = "UTF-8";
+
+        Path path;
+
+        if (input instanceof Path) {
+            path = (Path) input;
+        } else if (input instanceof File) {
+            path = ((File) input).toPath();
+        }
+
+        else {
+            try {
+                // Input corresponds to the parsed file
+                Map<String, Object> mapModel = createModel(input);
+                mapModel = (Map<String, Object>) mapModel.get("model");
+                path = Paths.get(mapModel.get("path").toString());
+                return isValidInput(path);
+            } catch (NullPointerException e) {
+                return false;
+            }
+
+        }
+
+        if (request.isNotConnected()) {
+            startServerConnection();
+        }
+
+        // File content is not needed, as only the file extension is checked
+        fileContents = new String("");
+        // String fileName = path.getFileName().toString();
+        String fileName = path.toString();
+        InputFileTo inputFile = new InputFileTo(fileName, fileContents, inputCharset);
+
+        HttpURLConnection conn = request.getConnection("POST", "Content-Type", "application/json", "isValidInput");
+
+        if (request.sendRequest(inputFile, conn, "UTF-8")) {
+
+            StringBuffer response = new StringBuffer();
+            try (InputStreamReader isr = new InputStreamReader(conn.getInputStream());
+                BufferedReader br = new BufferedReader(isr);) {
+
+                LOG.info("Receiving response from Server....");
+                Stream<String> s = br.lines();
+                s.parallel().forEachOrdered((String line) -> {
```

Comment:
Here I think we should not use `Stream`, as `isValidInput` just returns true or false, it seems to me too much overhead for only one line. The parallel stream is only useful when there are multiple lines. 

I think it would be better to just read directly the first line from `BufferedReader`.